lib/ApiPlatform/Serializer/InvalidSearchConditionNormalizer.php

```diff
@@ -27,53 +31,80 @@
 
-    public function __construct(UrlGeneratorInterface $urlGenerator)
+    public function __construct(array $serializePayloadFields = null, NameConverterInterface $nameConverter = null)
     {
-        $this->urlGenerator = $urlGenerator;
+        $this->nameConverter = $nameConverter;
+        $this->serializePayloadFields = $serializePayloadFields;
     }
 
     /**
-     * Normalizes an object into a set of arrays/scalars.
-     *
-     * @param InvalidSearchConditionException $object  object to normalize
-     * @param string                          $format  format the normalization result will be encoded as
-     * @param array                           $context Context options for the normalizer
-     *
-     * @return array
+     * @inheritdoc
      */
     public function normalize($object, $format = null, array $context = [])
     {
-        $violations = [];
-        $messages = [];
+        $list = new ConstraintViolationList();
 
+        /** @var ConditionErrorMessage $message */
         foreach ($object->getErrors() as $message) {
-            $violations[] = [
-                'propertyPath' => $message->path,
-                'message' => $message->message,
-            ];
-
-            $propertyPath = $message->path;
-            $prefix = $propertyPath ? sprintf('%s: ', $propertyPath) : '';
-
-            $messages[] = $prefix.$message->message;
+            $violation = new ConstraintViolation(
+                $message->message,
+                $message->messageTemplate,
+                $message->messageParameters,
+                null, // root
+                $message->path,
+                null, // invalidValue
+                $message->messagePluralization,
+                null, // code
+                null, // constraint
+                $message->cause
+            );
+            $list->add($violation);
         }
 
+        list($messages, $violations) = $this->getMessagesAndViolations($list);
+
         return [
-            '@context' => $this->urlGenerator->generate('api_jsonld_context', ['shortName' => 'ConstraintViolationList']),
-            '@type' => 'ConstraintViolationList',
-            'hydra:title' => $context['title'] ?? 'An error occurred',
-            'hydra:description' => $messages ? implode("\n", $messages) : (string) $object,
+            'type' => $context['type'] ?? 'https://tools.ietf.org/html/rfc2616#section-10',
+            'title' => $context['title'] ?? 'An error occurred',
+            'detail' => $messages ? implode("\n", $messages) : (string) $object,
             'violations' => $violations,
         ];
     }
 
-    public function supportsNormalization($data, $format = null)
+    public function supportsNormalization($data, $format = null): bool
     {
         return self::FORMAT === $format && $data instanceof InvalidSearchConditionException;
     }
+
+    /**
+     * @author Kévin Dunglas <dunglas@gmail.com>
+     *
+     * @see https://github.com/api-platform/core
```

Comment:
Added from an `@internal` abstract class which was only available since APIP 2.2, better to inline.